scipy/integrate/_bvp.py

```diff
@@ -1097,34 +1105,39 @@ def solve_bvp(fun, bc, x, y, p=None, S=None, fun_jac=None, bc_jac=None,
             status = 1
             if verbose == 2:
                 nodes_added = "({})".format(nodes_added)
-                print_iteration_progress(iteration, max_rms_res, m,
-                                         nodes_added)
+                print_iteration_progress(iteration, max_rms_res, max_bc_res,
+                                         m, nodes_added)
             break
 
         if verbose == 2:
-            print_iteration_progress(iteration, max_rms_res, m, nodes_added)
+            print_iteration_progress(iteration, max_rms_res, max_bc_res, m,
+                                     nodes_added)
 
         if nodes_added > 0:
             x = modify_mesh(x, insert_1, insert_2)
             h = np.diff(x)
             y = sol(x)
-        else:
+        elif max_bc_res <= bc_tol:
```

Comment:
I realized that this modification creates a potential for an infinite loop --- nodes are not added and the iterations continue. Hopefully `max_bc_res` will be reduced as necessary and the solver will terminate, however it is not guaranteed (imaging `bc_tol = 0`).

What if we allow only 1 additional iteration after the nodes were not added (`tol` satisfied)? If `bc_tol` is not satisfied after that, we terminate with another `status` (`3:  The solver wasn't able to satisfy boundary conditions tolerance`) (or something along these lines). Perhaps ideally we should add another parameter (`max_iterations` or something), but we already have `max_nodes` and I don't like polluting the list parameters to handle this rare case.

Reply:
@scbarton can we handle this issue somehow? Either discuss it or implement the modification I proposed.